template/downstream.c.erb

```diff
@@ -35,6 +35,31 @@ void btx_downstream_push_message(
   DL_APPEND(message_iterator_private_data->queue, elt);
 }
 
+// Exposed to the user
+void btx_push_message(void *btx_handle, const bt_message *message) {
+  _btx_push_message((btx_message_iterator_t *)btx_handle, message);
+}
+
+// User by everything
+void btx_downstream_push_message(
+    btx_message_iterator_t *message_iterator_private_data,
+    const bt_message *message) {
+
+  on_push_callback_f *p =
+      message_iterator_private_data->common_data->on_push_callback;
+  if (p)
+    p(((void *)message_iterator_private_data),
```

Comment:
Since `p` is a callback, should we pass always (void *)common_data, as in the other callbacks? This may keep some consistency from the user's point of view.

Reply:
For a user point of view, it's a void pointer. It's forbidden to do anything with it. 
.Saying that we pass `common_data`, it's just let natural but can be done.

Reply:
Ok we cannot as we need the `message_iterator_private_data` inside the `_btx_push_message`. And common_data is inside `message_iterator_private_data`